_posts/aws/2019-06-18-AWS-SecretsInWebhookStages.md

```diff
@@ -0,0 +1,64 @@
+---
+date: 2019-05-29
+title: AWS S3 Secrets in Custom Webhook Stages
+categories:
+   - AWS
+description: How to use secrets from S3 in custom webhook stages.
+type: Document
+---
+
+## Background
+
+Early this year from [version 1.15.0](https://github.com/spinnaker/halyard/releases), Halyard supports secret management that allow us to store our passwords even files as kube configs in S3 and reference them from a yml file, this is a great feature that we can use in several scenarios where we want to hide information and store them in a safe place.
+
+This document will show you how to use S3 secrets in a custom webhook stage.  This document also assumes that you have configured an AWS account.  If you need more instructions on configuring your AWS account please refer to either [Deploying to AWS from Spinnaker (using IAM instance roles)](https://docs.armory.io/spinnaker-install-admin-guides/add-aws-account-iam/) or [Deploying to AWS from Spinnaker (using IAM credentials)](https://docs.armory.io/spinnaker-install-admin-guides/add-aws-account/) 
```

Comment:
We can also use S3 secrets when doing deployments to Kubernetes, without actually having configured an AWS account. In this case, access to S3 bucket is generally done using IAM instance roles of the EKS cluster where Spinnaker is installed.
That said, this post can be made more generic for using any supported secrets engine (S3 in OSS and Vault in Armory Spinnaker, actual example in S3 is fine), and it doesn't have special cloud providers requirements because Webhook stage can be used with any provider. Then the category would be more fitted to `Pipelines` instead of `AWS`.